### ISSUE TITLE
Chat - Enter to save display name triggers mention suggestion

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29351]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.80.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1696919926668389

## Action Performed:
1. Open the app
2. Open any report
3. Type @ to trigger mention
4. Open settings->profile->display name
5. Make any changes and click enter to save display name
6. Observe that even though RHN is open, mention suggestion is triggered

## Expected Result:
App should not trigger mention suggestion on enter to save display name

## Actual Result:
App triggers mention suggestion on enter to save display name

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

</details>

<details>
<summary>iOS: Native</summary>

 

</details>

<details>
<summary>iOS: mWeb Safari</summary>

 

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/ba490622-c88b-41e8-91aa-619378e19b0b

https://github.com/Expensify/App/assets/93399543/fa4d3700-f2f4-4cce-8053-cc69b5fd723f

</details>

<details>
<summary>MacOS: Desktop</summary>

 

https://github.com/Expensify/App/assets/93399543/d9366047-50d4-4821-bbcf-63599a15d8db

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Updating profile data triggers mentions suggestion.

### What is the root cause of that problem?
Actually pressing `Enter` key is not required, notice that only if you update your display name it will trigger mentions. That is because `calculateMentionSuggestion`, which determines if suggestion should be display or not, is dependent on `personalDetails`. That is why when you update your display name it will trigger `calculateMentionSuggestion` function and all conditions are met to show mentions.

https://github.com/Expensify/App/blob/89ac222a2bbe6c8672559fa229e4686430e688b4/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L182-L233

### What changes do you think we should make in order to solve the problem?
I think the best approach is to prevent suggestion from displaying when composer is not focused. We can achieve this by passing `textInputRef.current.isFocused()` to `Suggestions` component as `isFocused` here:
https://github.com/Expensify/App/blob/f878fbd0f06ad7e6eb033a354b4140dfbc5fd3db/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L549
And pass it to `SuggestionMention` component from `Suggestions` here:
https://github.com/Expensify/App/blob/f9c43fc9f0580b467246f0dd3e76ffae0f43ce0f/src/pages/home/report/ReportActionCompose/Suggestions.js#L104

We can either add condition to here(I prefer this one since it prevents complicated calculations too) 
https://github.com/Expensify/App/blob/89ac222a2bbe6c8672559fa229e4686430e688b4/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L184

or here
https://github.com/Expensify/App/blob/89ac222a2bbe6c8672559fa229e4686430e688b4/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L270

### What alternative solutions did you explore? (Optional)

**Comment by deetergp** [comment]
I agree with @Ollyws, let's go with @alitoshmatov's very clear proposal. 👍

**Comment by alitoshmatov** [comment]
PR is ready - https://github.com/Expensify/App/pull/29951
